training-api/src/sso.ts

```diff
@@ -0,0 +1,14 @@
+import z from "zod";
+
+import { api } from "./common";
+
+const forumSsoSchema = z.object({
+  return_sso_url: z.string(),
+  parameters: z.string(),
+});
+
+export type ForumSso = z.infer<typeof forumSsoSchema>;
+
+export function getForumSso(payload: string, signature: string, cookie: string): Promise<ForumSso> {
```

Comment:
Should `cookie` be an optional argument?

Reply:
For the underlying `api()` call yes, but for `getForumSso` I would say no. The backend needs this token to identify the user: https://github.com/algorithm-ninja/cmsocial/blob/1cfff5b884d8bcbbeaf84ecef3f5b20e7133aeec/cmsocial/server/pws.py#L635-L637